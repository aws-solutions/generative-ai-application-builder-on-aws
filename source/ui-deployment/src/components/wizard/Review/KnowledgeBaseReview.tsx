/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/
import {
    Box,
    ColumnLayout,
    Container,
    Header,
    SpaceBetween,
    Button,
    ExpandableSection
} from '@cloudscape-design/components';
import { ReviewSectionProps } from '../interfaces/Steps';
import { DEFAULT_STEP_INFO, KNOWLEDGE_BASE_PROVIDERS, WIZARD_PAGE_INDEX } from '../steps-config';
import JsonCodeView from '@/components/commons/json-code-view';
import { ValueWithLabel } from '@/components/useCaseDetails/common-components';
import { scoreToKendraMapping } from '../KnowledgeBase/helpers';

interface KnowledgeBaseReviewProps extends ReviewSectionProps {
    knowledgeBaseData: any;
}

export const KnowledgeBaseReview = (props: KnowledgeBaseReviewProps) => {
    return (
        <SpaceBetween size="xs" data-testid="review-knowledge-base-container">
            <Header
                variant="h3"
                headingTagOverride="h2"
                actions={
                    <Button
                        className="edit-step-btn"
                        onClick={() => props.setActiveStepIndex(WIZARD_PAGE_INDEX.KNOWLEDGE_BASE)}
                    >
                        Edit
                    </Button>
                }
            >
                {props.header}
            </Header>

            <SpaceBetween size="l">
                <Container
                    header={
                        <Header variant="h2" headingTagOverride="h3">
                            Retrieval Augmented Generation (RAG)
                        </Header>
                    }
                >
                    <ColumnLayout columns={1} variant="text-grid">
                        <div>
                            <Box variant="awsui-key-label">
                                Do you want to enable Retrieval Augmented Generation (RAG) for this use case?
                            </Box>
                            <div>{props.knowledgeBaseData.isRagRequired ? 'Yes' : 'No'}</div>
                        </div>
                    </ColumnLayout>
                </Container>
                {props.knowledgeBaseData.isRagRequired && (
                    <Container
                        header={
                            <Header variant="h2" headingTagOverride="h3">
                                Knowledge base options
                            </Header>
                        }
                        footer={
                            props.knowledgeBaseData.existingKendraIndex === 'no' && (
                                <Box>
                                    {
                                        <ExpandableSection headerText="Additional options" variant="footer">
                                            <ColumnLayout columns={2} variant="text-grid">
                                                <ValueWithLabel label="Kendra edition">
                                                    {props.knowledgeBaseData.kendraEdition.label}
                                                </ValueWithLabel>

                                                <ValueWithLabel label="Kendra additional query capacity">
                                                    {props.knowledgeBaseData.kendraAdditionalQueryCapacity}
                                                </ValueWithLabel>

                                                <ValueWithLabel label="Kendra additional storage capacity">
                                                    {props.knowledgeBaseData.kendraAdditionalStorageCapacity}
                                                </ValueWithLabel>
                                            </ColumnLayout>
                                        </ExpandableSection>
                                    }
                                </Box>
                            )
                        }
                        data-testid="knowledge-base-options-container"
                    >
                        <ColumnLayout columns={2} variant="text-grid">
                            <ValueWithLabel label="Knowledge base type">
                                {props.knowledgeBaseData.knowledgeBaseType.label}
                            </ValueWithLabel>

                            {props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.kendra && (
                                <>
                                    <ValueWithLabel label="Do you have an existing Kendra index?">
                                        {props.knowledgeBaseData.existingKendraIndex}
                                    </ValueWithLabel>

                                    {props.knowledgeBaseData.existingKendraIndex === 'yes' && (
                                        <ValueWithLabel label="Kendra index ID">
                                            {props.knowledgeBaseData.kendraIndexId}
                                        </ValueWithLabel>
                                    )}

                                    {props.knowledgeBaseData.existingKendraIndex === 'no' && (
                                        <ValueWithLabel label="New Kendra index name">
                                            {props.knowledgeBaseData.kendraIndexName}
                                        </ValueWithLabel>
                                    )}
                                </>
                            )}

                            {props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.bedrock && (
                                <>
                                    <ValueWithLabel label="Bedrock Knowledge Base ID">
                                        {props.knowledgeBaseData.bedrockKnowledgeBaseId}
                                    </ValueWithLabel>
                                </>
                            )}
                        </ColumnLayout>
                    </Container>
                )}

                {props.knowledgeBaseData.isRagRequired && (
                    <Container
                        header={
                            <Header variant="h2" headingTagOverride="h3">
                                Advanced RAG configurations
                            </Header>
                        }
                        data-testid="advanced-rag-configs-container"
                    >
                        <ColumnLayout columns={2} variant="text-grid">
                            <ValueWithLabel label="Maximum number of documents">
                                {props.knowledgeBaseData.maxNumDocs}
                            </ValueWithLabel>

                            <ValueWithLabel label="Score Threshold">
                                {props.knowledgeBaseData.knowledgeBaseType.value === KNOWLEDGE_BASE_PROVIDERS.kendra
                                    ? scoreToKendraMapping(props.knowledgeBaseData.scoreThreshold)
                                    : props.knowledgeBaseData.scoreThreshold}
                            </ValueWithLabel>

                            <ValueWithLabel label="Static response when no documents found">
                                {props.knowledgeBaseData.noDocsFoundResponse
                                    ? props.knowledgeBaseData.noDocsFoundResponse
                                    : '-'}
                            </ValueWithLabel>

                            <ValueWithLabel label="Display document source">
                                {props.knowledgeBaseData.returnDocumentSource ? 'Yes' : 'No'}
                            </ValueWithLabel>

                            <ValueWithLabel label="Role based access control enabled?">
                                {props.knowledgeBaseData.enableRoleBasedAccessControl ? 'Yes' : 'No'}
                            </ValueWithLabel>

                            {props.knowledgeBaseData.bedrockOverrideSearchType && (
                                <ValueWithLabel label="Override Search Type">
                                    {props.knowledgeBaseData.bedrockOverrideSearchType.label}
                                </ValueWithLabel>
                            )}

                            {props.knowledgeBaseData.queryFilter !== DEFAULT_STEP_INFO.knowledgeBase.queryFilter && (
                                <div data-testid="query-filter-review">
                                    {props.knowledgeBaseData.knowledgeBaseType.value ===
                                        KNOWLEDGE_BASE_PROVIDERS.bedrock && (
                                        <Box variant="awsui-key-label">Retrieval filter</Box>
                                    )}

                                    {props.knowledgeBaseData.knowledgeBaseType.value ===
                                        KNOWLEDGE_BASE_PROVIDERS.kendra && (
                                        <Box variant="awsui-key-label">Attribute filter</Box>
                                    )}

                                    <JsonCodeView content={props.knowledgeBaseData.queryFilter} />
                                </div>
                            )}
                        </ColumnLayout>
                    </Container>
                )}
            </SpaceBetween>
        </SpaceBetween>
    );
};

export default KnowledgeBaseReview;
