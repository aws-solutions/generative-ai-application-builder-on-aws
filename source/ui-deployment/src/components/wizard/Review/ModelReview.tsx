/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import {
    Box,
    ColumnLayout,
    Container,
    Header,
    SpaceBetween,
    Button,
    ExpandableSection,
    AttributeEditor
} from '@cloudscape-design/components';
import { ReviewSectionProps } from '../interfaces/Steps';
import { MODEL_PROVIDER_NAME_MAP, WIZARD_PAGE_INDEX } from '../steps-config';
import { createBox, escapedNewLineToLineBreakTag, ValueWithLabel } from '../../useCaseDetails/common-components';
import { useComponentId } from '../../commons/use-component-id';

import { ModelParams } from '../Model/AdvancedModelSettings';
import { useQueryClient } from '@tanstack/react-query';

interface ModelReviewProps extends ReviewSectionProps {
    knowledgeBaseData: any;
    modelData: any;
}

/**
 * Removes any empty or invalid model params that are missing values for key, value, or type
 * from modelParameters list.
 * @param {Array} modelParameters Array of items from the AttributeEditor component in
 * step 2.
 */
const sanitizeModelParams = (modelParameters: ModelParams[]) => {
    return modelParameters
        .filter((param) => Object.keys(param).length !== 0)
        .filter((param) => param.key && param.value && param.type);
};

const CreateAttributeEditor = ({
    modelParameters,
    setActiveStepIndex
}: {
    modelParameters: ModelParams[];
    setActiveStepIndex: (index: number) => void;
}) => {
    const WIZARD_MODEL_STEP_INDEX = 2;

    return (
        <AttributeEditor
            onAddButtonClick={() => setActiveStepIndex(WIZARD_MODEL_STEP_INDEX)}
            items={modelParameters}
            addButtonText="Add new item"
            definition={[
                {
                    label: 'Key',
                    control: (item) => createBox(item.key)
                },
                {
                    label: 'Value',
                    control: (item) => createBox(item.value)
                },
                {
                    label: 'Type',
                    control: (item) => createBox(item.type.value)
                }
            ]}
            isItemRemovable={() => false}
        />
    );
};

export const ModelReview = (props: ModelReviewProps) => {
    const queryClient = useQueryClient();

    props.modelData.modelParameters = sanitizeModelParams(props.modelData.modelParameters);

    const invalidateQueryAndEdit = () => {
        props.setActiveStepIndex(WIZARD_PAGE_INDEX.MODEL);
        queryClient.invalidateQueries({
            queryKey: ['modelInfo', props.modelData.modelProvider.value, props.modelData.modelName]
        });
    };

    const componentId = useComponentId();

    return (
        <SpaceBetween size="xs" className="step-3-review">
            <Header
                variant="h3"
                headingTagOverride="h2"
                actions={<Button onClick={invalidateQueryAndEdit}>Edit</Button>}
            >
                {props.header}
            </Header>
            <SpaceBetween size="l">
                <Container
                    header={
                        <Header variant="h2" headingTagOverride="h3">
                            Model selection
                        </Header>
                    }
                    footer={
                        <ExpandableSection
                            headerText="Additional settings"
                            variant="footer"
                            defaultExpanded
                            data-testid="model-review-additional-settings-expandable-section"
                        >
                            <ColumnLayout columns={2} variant="text-grid">
                                <ValueWithLabel label="Model temperature">{props.modelData.temperature}</ValueWithLabel>
                                <ValueWithLabel label="Enable guardrails">
                                    {props.modelData.enableGuardrails ? 'yes' : 'no'}
                                </ValueWithLabel>
                                {props.modelData.enableGuardrails && (
                                    <>
                                        <ValueWithLabel label="Guardrail Identifier">
                                            {props.modelData.guardrailIdentifier}
                                        </ValueWithLabel>
                                        <ValueWithLabel label="Guardrail Version">
                                            {props.modelData.guardrailVersion}
                                        </ValueWithLabel>
                                    </>
                                )}
                                <ValueWithLabel label="Verbose">
                                    {props.modelData.verbose ? 'on' : 'off'}
                                </ValueWithLabel>
                                <ValueWithLabel label="Streaming">
                                    {props.modelData.streaming ? 'on' : 'off'}
                                </ValueWithLabel>
                            </ColumnLayout>
                        </ExpandableSection>
                    }
                    data-testid="review-model-details-container"
                >
                    <ColumnLayout columns={2} variant="text-grid">
                        <ValueWithLabel label="Model provider">{props.modelData.modelProvider.label}</ValueWithLabel>

                        {props.modelData.modelProvider.value !== MODEL_PROVIDER_NAME_MAP.SageMaker && (
                            <ValueWithLabel label="Model name">{props.modelData.modelName}</ValueWithLabel>
                        )}

                        {props.modelData.provisionedModel && (
                            <ValueWithLabel label="Model ARN">{props.modelData.modelArn}</ValueWithLabel>
                        )}

                        {props.modelData.modelProvider.value === MODEL_PROVIDER_NAME_MAP.SageMaker && (
                            <ValueWithLabel label="SageMaker endpoint name">
                                {props.modelData.sagemakerEndpointName}
                            </ValueWithLabel>
                        )}
                        {props.modelData.modelProvider.value === MODEL_PROVIDER_NAME_MAP.SageMaker && (
                            <ValueWithLabel label="SageMaker output path schema">
                                {props.modelData.sagemakerOutputSchema}
                            </ValueWithLabel>
                        )}
                        {props.modelData.modelProvider.value === MODEL_PROVIDER_NAME_MAP.SageMaker && (
                            <ValueWithLabel label="SageMaker input schema">
                                <Box variant="code">
                                    {escapedNewLineToLineBreakTag(props.modelData.sagemakerInputSchema, componentId)}
                                </Box>
                            </ValueWithLabel>
                        )}
                    </ColumnLayout>
                </Container>
                {props.modelData.modelParameters.length > 0 && (
                    <Container
                        header={
                            <Header variant="h2" headingTagOverride="h3">
                                Advanced model parameters
                            </Header>
                        }
                    >
                        <CreateAttributeEditor
                            modelParameters={props.modelData.modelParameters}
                            setActiveStepIndex={props.setActiveStepIndex}
                        />
                    </Container>
                )}
            </SpaceBetween>
        </SpaceBetween>
    );
};

export default ModelReview;
