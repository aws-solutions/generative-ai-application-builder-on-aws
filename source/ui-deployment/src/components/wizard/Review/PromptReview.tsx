import { Button, Container, Header, SpaceBetween } from '@cloudscape-design/components';
import { PromptDetails } from '@/components/useCaseDetails/PromptDetails';
import { ReviewSectionProps } from '../interfaces/Steps';
import { createConversationMemoryApiParams, createLLMParamsApiParams } from '../utils';
import { WIZARD_PAGE_INDEX } from '../steps-config';

interface PromptReviewProps extends ReviewSectionProps {
    modelData: any;
    promptData: any;
    isRag: boolean;
    'data-testid': string;
}

export const PromptReview = (props: PromptReviewProps) => {
    const llmParamsPayload = createLLMParamsApiParams(props.modelData, props.promptData, props.isRag);
    const memoryParamsPayload = createConversationMemoryApiParams(props.promptData);
    const deployment = {
        ...llmParamsPayload,
        ...memoryParamsPayload
    };

    return (
        <SpaceBetween size="xs" data-testid={props['data-testid']}>
            <Header
                variant="h3"
                actions={<Button onClick={() => props.setActiveStepIndex(WIZARD_PAGE_INDEX.PROMPT)}>Edit</Button>}
            >
                {props.header}
            </Header>
            <Container>
                <PromptDetails selectedDeployment={deployment} />
            </Container>
        </SpaceBetween>
    );
};

export default PromptReview;
