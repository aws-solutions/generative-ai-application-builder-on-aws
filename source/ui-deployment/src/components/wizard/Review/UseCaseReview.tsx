/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import { Box, ColumnLayout, Container, Header, SpaceBetween, Button } from '@cloudscape-design/components';
import { ReviewSectionProps } from '../interfaces/Steps';
import { WIZARD_PAGE_INDEX } from '../steps-config';
import { ValueWithLabel } from '@/components/useCaseDetails/common-components';

interface UseCaseReviewProps extends ReviewSectionProps {
    useCaseData: any;
}

export const UseCaseReview = (props: UseCaseReviewProps) => {
    return (
        <SpaceBetween size="xs">
            <Header
                variant="h3"
                headingTagOverride="h2"
                actions={<Button onClick={() => props.setActiveStepIndex(WIZARD_PAGE_INDEX.USE_CASE)}>Edit</Button>}
            >
                {props.header}
            </Header>
            <Container
                header={
                    <Header variant="h2" headingTagOverride="h3">
                        Use case options
                    </Header>
                }
                data-testid="review-use-case-details-container"
            >
                <ColumnLayout columns={2} variant="text-grid" data-testid="review-use-case-details">
                    <ValueWithLabel label="Use case">{props.useCaseData.useCase.label}</ValueWithLabel>

                    <ValueWithLabel label="Use case name">{props.useCaseData.useCaseName}</ValueWithLabel>

                    {props.useCaseData.defaultUserEmail !== '' && (
                        <ValueWithLabel label="Use case email">{props.useCaseData.defaultUserEmail}</ValueWithLabel>
                    )}

                    {props.useCaseData.useCaseDescription !== '' && (
                        <ValueWithLabel label="Use case description">
                            {props.useCaseData.useCaseDescription}
                        </ValueWithLabel>
                    )}

                    <ValueWithLabel label="Deploy UI">{props.useCaseData.deployUI ? 'Yes' : 'No'}</ValueWithLabel>
                </ColumnLayout>
            </Container>
        </SpaceBetween>
    );
};

export default UseCaseReview;
