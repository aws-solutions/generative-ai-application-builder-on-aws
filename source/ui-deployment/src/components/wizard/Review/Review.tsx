/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { Box, SpaceBetween } from '@cloudscape-design/components';

import UseCaseReview from './UseCaseReview';
import { ReviewProps } from '../interfaces/Steps';
import { KnowledgeBaseReview } from './KnowledgeBaseReview';
import { ModelReview } from './ModelReview';
import VpcConfigReview from './VpcConfigReview';
import PromptReview from './PromptReview';

export const Review = ({ info: { useCase, knowledgeBase, model, vpc, prompt }, setActiveStepIndex }: ReviewProps) => {
    return (
        <Box margin={{ bottom: 'l' }} data-testid="review-deployment-component">
            <SpaceBetween size="xxl">
                <UseCaseReview
                    header="Step 1: Use case"
                    useCaseData={useCase}
                    setActiveStepIndex={setActiveStepIndex}
                />

                <VpcConfigReview
                    header="Step 2: VPC Configuration"
                    vpcData={vpc}
                    setActiveStepIndex={setActiveStepIndex}
                />

                <ModelReview
                    header="Step 3: Model"
                    knowledgeBaseData={knowledgeBase}
                    modelData={model}
                    setActiveStepIndex={setActiveStepIndex}
                />

                <KnowledgeBaseReview
                    header="Step 4: Knowledge base"
                    knowledgeBaseData={knowledgeBase}
                    setActiveStepIndex={setActiveStepIndex}
                />

                <PromptReview
                    header="Step 5: Prompt"
                    modelData={model}
                    promptData={prompt}
                    isRag={knowledgeBase.isRagRequired}
                    setActiveStepIndex={setActiveStepIndex}
                    data-testid="review-prompt-container"
                />
            </SpaceBetween>
        </Box>
    );
};

export default Review;
