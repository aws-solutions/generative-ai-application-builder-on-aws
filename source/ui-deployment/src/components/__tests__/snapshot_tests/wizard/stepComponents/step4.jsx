/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 **********************************************************************************************************************/

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import HomeContext from '../../../../../home/home.context';
import Review from '../../../../wizard/stepComponents/step4';
import {
    DEFAULT_STEP_INFO,
    USE_CASE_OPTIONS,
    KNOWLEDGE_BASE_TYPES,
    DEFAULT_ADDITIONAL_KENDRA_QUERY_CAPACITY,
    DEFAULT_ADDITIONAL_KENDRA_STORAGE_CAPACITY,
    KENDRA_EDITIONS,
    DEFAULT_KENDRA_NUMBER_OF_DOCS,
    MODEL_FAMILY_PROVIDER_OPTIONS
} from '../../../../wizard/steps-config';

jest.mock('@cloudscape-design/components', () => {
    const Components = jest.genMockFromModule('@cloudscape-design/components');
    for (const componentName of Object.keys(Components)) {
        Components[componentName] = componentName;
    }
    return Components;
});

describe('Chat', () => {
    const contextValue = {
        'dispatch': jest.fn(),
        'state': {
            'selectedDeployment': {},
            'deploymentsData': [],
            'deploymentAction': 'CREATE',
            'authorized': true
        }
    };

    const steps_info = {
        useCase: {
            useCase: USE_CASE_OPTIONS[0],
            useCaseName: 'fake-use-case',
            useCaseDescription: 'fake-description',
            defaultUserEmail: 'fake_email@example.com',
            inError: false
        },
        knowledgeBase: {
            isRagRequired: true,
            knowledgeBaseType: KNOWLEDGE_BASE_TYPES[0],
            existingKendraIndex: 'no',
            kendraIndexId: '',
            kendraAdditionalQueryCapacity: DEFAULT_ADDITIONAL_KENDRA_QUERY_CAPACITY,
            kendraAdditionalStorageCapacity: DEFAULT_ADDITIONAL_KENDRA_STORAGE_CAPACITY,
            kendraEdition: KENDRA_EDITIONS[0],
            maxNumDocs: DEFAULT_KENDRA_NUMBER_OF_DOCS,
            inError: false,
            kendraIndexName: 'fake-index'
        },
        model: {
            modelProvider: MODEL_FAMILY_PROVIDER_OPTIONS[0],
            apiKey: 'fake-key',
            modelName: 'fake-name',
            modelFamily: 'Bedrock',
            promptTemplate: 'fake-template',
            inferenceEndpoint: 'fake-inference-endpoint',
            modelParameters: [],
            inError: false,
            temperature: 0.1,
            verbose: false,
            streaming: false
        }
    };

    test('Snapshot test for wizard step 4', async () => {
        const tree = renderer
            .create(
                <>
                    <HomeContext.Provider
                        value={{
                            ...contextValue
                        }}
                    >
                        <Review info={steps_info} setActiveStepIndex={jest.fn()} />
                    </HomeContext.Provider>
                </>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
