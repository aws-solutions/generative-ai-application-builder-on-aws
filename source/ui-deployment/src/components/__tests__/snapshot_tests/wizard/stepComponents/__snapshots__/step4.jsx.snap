// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Snapshot test for wizard step 4 1`] = `
<Box
  data-testid="review-deployment-component"
  margin={
    Object {
      "bottom": "l",
    }
  }
>
  <SpaceBetween
    size="xxl"
  >
    <SpaceBetween
      className="step-1-review"
      size="xs"
    >
      <Header
        actions={
          <Button
            className="edit-step-btn"
            onClick={[Function]}
          >
            Edit
          </Button>
        }
        headingTagOverride="h2"
        variant="h3"
      >
        Step 1: Use case
      </Header>
      <Container
        header={
          <Header
            headingTagOverride="h3"
            variant="h2"
          >
            Use case options
          </Header>
        }
      >
        <ColumnLayout
          columns={2}
          data-testid="review-use-case-details"
          variant="text-grid"
        >
          <div>
            <Box
              variant="awsui-key-label"
            >
              Use case
            </Box>
            <div>
              Chat
            </div>
          </div>
          <div>
            <Box
              variant="awsui-key-label"
            >
              Use case name
            </Box>
            <div>
              fake-use-case
            </div>
          </div>
          <div>
            <Box
              variant="awsui-key-label"
            >
              Use case email
            </Box>
            <div>
              fake_email@example.com
            </div>
          </div>
          <div>
            <Box
              variant="awsui-key-label"
            >
              Use case description
            </Box>
            <div>
              fake-description
            </div>
          </div>
        </ColumnLayout>
      </Container>
    </SpaceBetween>
    <SpaceBetween
      className="step-3-review"
      size="xs"
    >
      <Header
        actions={
          <Button
            className="edit-step-btn"
            onClick={[Function]}
          >
            Edit
          </Button>
        }
        headingTagOverride="h2"
        variant="h3"
      >
        Step 2: Model
      </Header>
      <SpaceBetween
        size="l"
      >
        <Container
          footer={
            <ExpandableSection
              defaultExpanded={true}
              headerText="Additional settings"
              variant="footer"
            >
              <ColumnLayout
                columns={2}
                variant="text-grid"
              >
                <div>
                  <Box
                    variant="awsui-key-label"
                  >
                    Model temperature
                  </Box>
                  <div>
                    0.1
                  </div>
                </div>
                <div>
                  <Box
                    variant="awsui-key-label"
                  >
                    Verbose
                  </Box>
                  <div>
                    off
                  </div>
                </div>
                <div>
                  <Box
                    variant="awsui-key-label"
                  >
                    Streaming
                  </Box>
                  <div>
                    off
                  </div>
                </div>
                <div>
                  <Box
                    variant="awsui-key-label"
                  >
                    System prompt
                  </Box>
                  <Box
                    data-testid="review-system-prompt"
                    variant="code"
                  >
                    fake-template
                  </Box>
                </div>
              </ColumnLayout>
            </ExpandableSection>
          }
          header={
            <Header
              headingTagOverride="h3"
              variant="h2"
            >
              Model selection
            </Header>
          }
        >
          <ColumnLayout
            columns={2}
            variant="text-grid"
          >
            <div>
              <Box
                variant="awsui-key-label"
              >
                Model provider
              </Box>
              <div>
                Bedrock
              </div>
            </div>
            <div>
              <Box
                variant="awsui-key-label"
              >
                Model name
              </Box>
              <div>
                fake-name
              </div>
            </div>
          </ColumnLayout>
        </Container>
      </SpaceBetween>
    </SpaceBetween>
    <SpaceBetween
      className="step-2-review"
      size="xs"
    >
      <Header
        actions={
          <Button
            className="edit-step-btn"
            onClick={[Function]}
          >
            Edit
          </Button>
        }
        headingTagOverride="h2"
        variant="h3"
      >
        Step 3: Knowledge base
      </Header>
      <SpaceBetween
        size="l"
      >
        <Container
          header={
            <Header
              headingTagOverride="h3"
              variant="h2"
            >
              Retrieval Augmented Generation (RAG)
            </Header>
          }
        >
          <ColumnLayout
            columns={1}
            variant="text-grid"
          >
            <div>
              <Box
                variant="awsui-key-label"
              >
                Do you want to enable Retrieval Augmented Generation (RAG) for this use case?
              </Box>
              <div>
                Yes
              </div>
            </div>
          </ColumnLayout>
        </Container>
        <Container
          footer={
            <Box>
              <ExpandableSection
                headerText="Additional options"
                variant="footer"
              >
                <ColumnLayout
                  columns={2}
                  variant="text-grid"
                >
                  <div>
                    <Box
                      variant="awsui-key-label"
                    >
                      Kendra additional query capacity
                    </Box>
                    <div />
                  </div>
                  <div>
                    <Box
                      variant="awsui-key-label"
                    >
                      Kendra additional storage capacity
                    </Box>
                    <div />
                  </div>
                  <div>
                    <Box
                      variant="awsui-key-label"
                    >
                      Kendra edition
                    </Box>
                    <div>
                      Developer
                    </div>
                  </div>
                </ColumnLayout>
              </ExpandableSection>
            </Box>
          }
          header={
            <Header
              headingTagOverride="h3"
              variant="h2"
            >
              Knowledge base options
            </Header>
          }
        >
          <ColumnLayout
            columns={2}
            variant="text-grid"
          >
            <div>
              <Box
                variant="awsui-key-label"
              >
                Knowledge base type
              </Box>
              <div>
                Kendra
              </div>
            </div>
            <div>
              <Box
                variant="awsui-key-label"
              >
                Do you have an existing Kendra index?
              </Box>
              <div>
                no
              </div>
            </div>
            <div>
              <Box
                variant="awsui-key-label"
              >
                New Kendra index name
              </Box>
              <div>
                fake-index
              </div>
            </div>
          </ColumnLayout>
        </Container>
        <Container
          header={
            <Header
              headingTagOverride="h3"
              variant="h2"
            >
              Advanced RAG configurations
            </Header>
          }
        >
          <ColumnLayout
            columns={2}
            variant="text-grid"
          >
            <div>
              <Box
                variant="awsui-key-label"
              >
                Maximum number of documents
              </Box>
              <div />
            </div>
          </ColumnLayout>
        </Container>
      </SpaceBetween>
    </SpaceBetween>
  </SpaceBetween>
</Box>
`;
